cmake/FindLIBHWLOC.cmake

```diff
@@ -14,28 +14,28 @@ find_file(LIBHWLOC_HEADER NAMES hwloc.h)
 get_filename_component(LIBHWLOC_INCLUDE_DIR ${LIBHWLOC_HEADER} DIRECTORY)
 set(LIBHWLOC_INCLUDE_DIRS ${LIBHWLOC_INCLUDE_DIR})
 
-if(LIBHWLOC_INCLUDE_DIR)
-    if(EXISTS "${LIBHWLOC_INCLUDE_DIR}/hwloc/autogen/config.h")
-        file(
-            STRINGS "${LIBHWLOC_INCLUDE_DIR}/hwloc/autogen/config.h"
-            LIBHWLOC_VERSION
-            REGEX
-                "#define[ \t]HWLOC_VERSION[ \t]\"([0-9]+.[0-9]+.[0-9]+(rc[0-9])?)(-git)?\""
-        )
-        string(
-            REGEX
-            REPLACE
-                "#define[ \t]HWLOC_VERSION[ \t]\"([0-9]+.[0-9]+.[0-9]+(rc[0-9])?)(-git)?\""
-                "\\1"
-                LIBHWLOC_VERSION
-                "${LIBHWLOC_VERSION}")
-    else()
-        message(
-            WARNING
-                "LIBHWLOC_INCLUDE_DIR found, but header with version info is missing"
-        )
-    endif()
-endif()
+set(HWLOC_VERSION_CODE
+    "
+#include <stdio.h>
+#include <stdlib.h>
+#include \"hwloc.h\"
+
+void main(int argc, char** argv) {
+    unsigned LIBHWLOC_API_PATCH = HWLOC_API_VERSION & 0xFF;
+    unsigned LIBHWLOC_API_MINOR = (HWLOC_API_VERSION >> 8) & 0xFF;
+    unsigned LIBHWLOC_API_MAJOR = (HWLOC_API_VERSION >> 16) & 0xFF;
+    printf(\"%d.%d.%d\", LIBHWLOC_API_MAJOR, LIBHWLOC_API_MINOR, LIBHWLOC_API_PATCH);
+}")
+
+set(HWLOC_VERSION_CODE_FILENAME "hwloc_get_version.c")
+file(WRITE "${CMAKE_BINARY_DIR}/${HWLOC_VERSION_CODE_FILENAME}"
```

Comment:
why not just create the .c file in our repo? Current dir is ok for storing this.
You could then copy this file (from cmake_current_source_dir to cmake_binary_dir) using `configure_file(<input> <output> COPYONLY)`

Reply:
> why not just create the .c file in our repo? Current dir is ok for storing this. You could then copy this file (from cmake_current_source_dir to cmake_binary_dir) using `configure_file(<input> <output> COPYONLY)`

is this a blocking comment? IMO having this code in cmake is fine, as it is only required for configuration/building, which is what cmake is for